### ISSUE TITLE
Java Mtags fails to index ParserConstants.java in the JDK

### ISSUE THREAD

**Issue by kiviok** [scalameta/metals#280]
```
ERROR s.m.m.s.InMemorySymbolIndex - Failed to index jar:file:///Library/Java/JavaVirtualMachines/jdk1.8.0_152.jdk/Contents/Home/src.zip!/com/sun/jmx/snmp/IPAcl/ParserConstants.java
java.lang.IllegalArgumentException: 1803 is not a valid offset, allowed [0..1801]
	at org.langmeta.internal.inputs.InternalInput.offsetToLine(InternalInput.scala:42)
	at org.langmeta.internal.inputs.InternalInput.offsetToLine$(InternalInput.scala:36)
	at org.langmeta.inputs.Input$VirtualFile.offsetToLine(Input.scala:74)
	at org.langmeta.inputs.Position$Range.endLine(Position.scala:40)
	at org.langmeta.languageserver.InputEnrichments$XtensionInputOffset$.toIndexRange$extension(InputEnrichments.scala:32)
	at scala.meta.metals.search.InMemorySymbolIndex.$anonfun$indexDocument$1(InMemorySymbolIndex.scala:193)
	at scala.meta.metals.search.InMemorySymbolIndex.$anonfun$indexDocument$1$adapted(InMemorySymbolIndex.scala:181)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	at scala.collection.Iterator.foreach$(Iterator.scala:929)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.meta.metals.search.InMemorySymbolIndex.indexDocument(InMemorySymbolIndex.scala:181)
	at scala.meta.metals.search.InMemorySymbolIndex.$anonfun$indexDatabase$1(InMemorySymbolIndex.scala:158)
	at scala.collection.Iterator.foreach(Iterator.scala:929)
	at scala.collection.Iterator.foreach$(Iterator.scala:929)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1417)
	at scala.collection.IterableLike.foreach(IterableLike.scala:71)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at scala.meta.metals.search.InMemorySymbolIndex.indexDatabase(InMemorySymbolIndex.scala:157)
	at scala.meta.metals.search.InMemorySymbolIndex.$anonfun$indexDependencyClasspath$7(InMemorySymbolIndex.scala:144)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at scala.meta.metals.search.InMemorySymbolIndex.$anonfun$indexDependencyClasspath$6(InMemorySymbolIndex.scala:137)
	at scala.meta.metals.search.InMemorySymbolIndex.$anonfun$indexDependencyClasspath$6$adapted(InMemorySymbolIndex.scala:136)
	at scala.meta.metals.storage.LevelDBMap$.withDB(LevelDBMap.scala:85)
	at scala.meta.metals.search.InMemorySymbolIndex.$anonfun$indexDependencyClasspath$1(InMemorySymbolIndex.scala:136)
	at monix.eval.internal.TaskRunLoop$.monix$eval$internal$TaskRunLoop$$loop$1(TaskRunLoop.scala:187)
	at monix.eval.internal.TaskRunLoop$RestartCallback$1.onSuccess(TaskRunLoop.scala:119)
	at monix.eval.Task$.$anonfun$forkedUnit$2(Task.scala:1463)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

I also do not have auto-completion. Where should I set the option `scalac.completions.enabled=true`

using sbt-metals 0.1.0-M1+71-8b2e169a and metals from code marketplace

**Comment by gabro** [comment]
Hi Pavel, thanks for trying Metals.

The first issue you have is related to indexing of Java symbols, which is being touched significantly by #275, so I suggest you try again after 0.1.0 is published officially.

As per auto-completion, the options refer to your VSCode settings (CMD + , on macOS).

Anyway, the current release is not intended for general use, and to further clarify this we've removed the extension from the Marketplace (we've erroneously left it published during some tests last week)

If you're interested in contributing, check out our [guidelines here](https://github.com/scalameta/metals/blob/master/CONTRIBUTING.md).

We expect to publish a 0.1.0 in the upcoming weeks, but we don't have an official timeline, since Metals is based entirely on voluntary efforts :)

**Comment by olafurpg** [comment]
I have hit this error too while indexing a com/sun/jmx/snmp/IPAcl/ParserConstants.java from the JDK. There seems to be a problem with how positions are computed from the qdox library we use to index outlines of Java sources.

A good place to test a fix would be here 
https://github.com/scalameta/metals/blob/905bf282299f7c136b40173103cf367073dc55f8/metals/src/test/scala/tests/mtags/JavaMtagsTest.scala#L159

**Comment by olafurpg** [comment]
I suspect the problem lies in this method here

https://github.com/scalameta/metals/blob/905bf282299f7c136b40173103cf367073dc55f8/metals/src/main/scala/scala/meta/metals/mtags/JavaMtags.scala#L56

**Comment by kiviok** [comment]
Gabriele, thank you.
Unfortunately i'm not a professional developer, only a hobbyist, so I can hardly help in essence.
But I will observe the development and participate in it as far as possible.

**Comment by melekhove** [comment]
Pull #325 fixes this issue, though the real problem is that qdox gives wrong line number 109 for some symbols ParserConstants.tokenImage.
I suppose that issue #86 has the same root cause, though it might be fixed by the mentioned pull as well.

**Comment by olafurpg** [comment]
Fixed in #325, thank @melekhove !
